src/conversion_helper.js

```diff
@@ -1,20 +1,22 @@
 const rh = require('./regex_helper');
+let inRangeVal = -Infinity;
```

Comment:
I'm not too sure about this being a global state variable. I see how it is making the code easier to write, but it's also creating unintentional "links" between distanceMap and weightMap for instance, which is weird. I would like to keep functions as pure and "side-effect free" as possible, is it possible to get rid of this?

Reply:
Well, really the only other way I see is to push this in the function, which is even more convoluted to trace back to source. IMO, a global value is also better as in future, we can expand to >2 values as well this way. One other way might be to make completely separate function for ranges, but that'll be very repetitive and make maintenance more complicated IMO.